sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClientIntegrationTest.java

```diff
@@ -54,4 +54,28 @@ void receiveMessageAutoComplete() {
             })
             .verifyComplete();
     }
+
+    /**
+     * Verifies that we can send and peek a message.
+     */
+    @Test
+    void peekMessage() {
+        // Arrange
+        final String messageId = UUID.randomUUID().toString();
+        final String contents = "Some-contents";
+        final ServiceBusMessage message = TestUtils.getServiceBusMessage(contents, messageId, 0);
+        final ReceiveMessageOptions options = new ReceiveMessageOptions().setAutoComplete(true);
+        receiver = createBuilder()
+            .receiveMessageOptions(options)
+            .buildAsyncReceiverClient();
+
+        // Assert & Act
+        StepVerifier.create(sender.send(message).thenMany(receiver.peek()))
```

Comment:
sender.send(message).then(receiver.peek()) is probably what you want. thenMany suggests it returns a flux.

Reply:
Good catch...